sorting/bubble_sort_recursion.c

```diff
@@ -0,0 +1,58 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+/**
+ * Swapped two numbers using pointer
+ * @param first first pointer of first number
+ * @param second second pointer of second number
+ */
+void swap(int *first, int *second)
+{
+    int temp = *first;
+    *first = *second;
+    *second = temp;
+}
+
+/**
+ * Bubble sort algorithm implements using recursion
+ * @param arr array to be sorted
+ * @param size size of array
+ */
+void bubbleSort(int *arr, int size)
+{
+    if (size == 1)
+    {
+        return;
+    }
+    bool swapped = false;
+    for (int i = 0; i < size - 1; ++i)
+    {
+        if (arr[i] > arr[i + 1])
+        {
+            swap(arr + i, arr + i + 1);
+            swapped = true;
+        }
+    }
+    if (swapped)
+    {
+        bubbleSort(arr, size - 1);
+    }
+}
+
```

Comment:
Please add a `test()` function that, uses `assert` statements to validate the implementation on random inputs.
You can use a simple loop like:
```c
for(int j=0; j < N-1; j++)
    assert(arr[j] < arr[j+1]);
```

Reply:
@kvedala Please review again

Reply:
A Test function should be atomic. 
Please see the test functions in other files [like this](https://github.com/TheAlgorithms/C/blob/5d9bf078514f3b5dd913100f10fe56f050f005d7/numerical_methods/qr_eigen_values.c#L224)

Reply:
updated. Please review again. Thanks :)